### PR TITLE
cc2538: fix radio driver issue which would impact device acting as SED

### PR DESCRIPTION
**Pull request by librasungirl**
The typo in otPlatRadioSleep() is the root cause for  issue #252, explained in #252

Hi @jwhui , please have a review.

### CODE REVIEW

File: examples/platforms/cc2538/radio.c [hunk 1]
```diff
@@ -103,6 +103,11 @@ void enableReceiver(void)
 
 void disableReceiver(void)
 {
+    if (!sIsReceiverEnabled)
```
Comment (jwhui) [inline examples/platforms/cc2538/radio.c]: Is this necessary?  If it is necessary, would prefer to either use `VerifyOrExit` or put the code below within the if-statement as is done in `enableReceiver`.  I'd like to maintain the code style of having only a single exit point for each method/function.

**Comment by jwhui** [comment]
Good catch!  Just one minor comment inline.

File: examples/platforms/cc2538/radio.c (hunk 1 above)
Comment (librasungirl) [inline examples/platforms/cc2538/radio.c]: adding it or not didn't have any influence on current test result, however I think adding it here would not do any harm. 
Updated a new patch which put it within if-statement.

**Comment by jwhui** [comment]
LGTM 👍 

Fixes #252.

**Pull request merged**